### ISSUE TITLE
OVER_QUEY_LIMIT Issue

### ISSUE THREAD

**Issue by ghostcrawler1** [iam4x/pokemongo-webspoof#435]
Hello,

Thank you for this app, its been working great until today when I tried to run a location search and I've been receiving the error OVER_QUEY_LIMIT. Any ideas how I can resolve this issue.

Thanks

**Comment by tonynkh** [comment]
Have the same issue

**Comment by OPaquay** [comment]
Same issue

**Comment by ItaloFSS** [comment]
Same issue

**Comment by adamkarb** [comment]
Same issue, likely we'll have to go change some google keys manually

**Comment by jerme404** [comment]
Yep, solve this by adding your own API key from https://developers.google.com/maps/documentation/javascript/get-api-key

**Comment by adamkarb** [comment]
Thanks!  Why did it all crap out at once I wonder?

**Comment by jnrepo** [comment]
reached api call limit

**Comment by lswlhn** [comment]
I've adde the following as @jerme404 instructed under GoogleMap in the index.js file, and replaced API_KEY with my own, still won't work. Any idea why? Thanks.
`bootstrapURLKeys={{
    key: API_KEY,
    language: 'ru',
    ...otherUrlParams,
  }}`

**Comment by jerme404** [comment]
I actually didn't use the recommended bootstrapURLKeys, I used the deprecated apiKey, like this

```
<GoogleMap
            ref={ (ref) => { this.map = ref } }
            zoom={ settings.zoom.get() }
            center={ [ latitude, longitude ] }
            onClick={ this.handleClick }
            options={ () => this.mapOptions }
            onGoogleApiLoaded={ this.handleGoogleMapLoaded }
            yesIWantToUseGoogleMapApiInternals={ true }
            apiKey={ MapsApi.apiKey }>
```

Where MapsApi.apiKey is in a configuration file I added, but you can just put in a string like

apiKey={ 'YOUR_API_KEY_HERE' }

I'll submit a PR with my changes in a bit.

**Comment by mrh3yd** [comment]
I just tried jerme404's solution; mine now looks like this:

```
<GoogleMap
            ref={ (ref) => { this.map = ref } }
            zoom={ settings.zoom.get() }
            center={ [ latitude, longitude ] }
            onClick={ this.handleClick }
            options={ () => this.mapOptions }
            onGoogleApiLoaded={ this.handleGoogleMapLoaded }
            yesIWantToUseGoogleMapApiInternals={ true }
            apiKey={ 'MY_API_KEY' }>
```

And still no work. Sad face. Any suggestions? I have a 7-year-old champing at the bit to catch some pocket monsters.

**Comment by jerme404** [comment]
Did you put an actual API key in there?

**Comment by lytesenthedarc** [comment]
I'm having the same issue. My code currently looks the same as yours, with apiKey={ 'YOUR_API_KEY_HERE' }

YOUR_API_KEY_HERE has been replaced with an api key generated by 'google maps apis' website. Still not working unfortunately.

**Comment by jerme404** [comment]
Still the same error OVER_QUEY_LIMIT after rebuilding?

**Comment by lytesenthedarc** [comment]
Yes

**Comment by lytesenthedarc** [comment]
should I upload my index file for review?

**Comment by mrh3yd** [comment]
Yes, I definitely pasted in the API key I got from Google in place of MY_API_KEY

Is there a step I'm missing? I edited the file, made the above changes, saved the file, and re-opened the spoof app.

**Comment by BearinErin** [comment]
I also followed the instructions for adding a new API and it did not fix it.

**Comment by mrh3yd** [comment]
I would also be happy to upload or email my index.js file, or use your recommended steps of creating another config file; I'm not 100% sure how to do the latter.

**Comment by lytesenthedarc** [comment]
Here's a copy of my non-functioning file
[index.js.zip](https://github.com/iam4x/pokemongo-webspoof/files/1272490/index.js.zip)

**Comment by rlam3** [comment]
Same issue. 

Does anybody know if using gitter to collaborate right now to get things working by tonight.

https://gitter.im

**Comment by jerme404** [comment]
You have to actually rebuild the project, it won't work by just editing the index.js file.

**Comment by lytesenthedarc** [comment]
Pardon my ignorance, is that the same as restarting the program, or is there a rebuild process we should be doing. And thank you very much for your help on this.

**Comment by jerme404** [comment]
If you navigate to the project in a terminal window, just type `npm run build` to build a new version.  That will create a new folder (if it's not there already) called 'pokemongo-webspoof-build' and your new build will be in there.

**Comment by mrh3yd** [comment]
Yep, that doesn't so much work for me either. Here's a screenshot from Terminal:

[https://ibb.co/i8O1jF](url)

Sorry to be such a pain...

**Comment by yrchern** [comment]
Simply open "GoogleMap.spec.js", find

const API_KEY = 'API_KEY';

and replace API_KEY with yours.

**Comment by ItaloFSS** [comment]
Im getting the following error while building. Any ideas?

Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration.output.path: The provided value "./dist" is not an absolute path!

**Comment by jerme404** [comment]
In webpack.config.babel, change the output path like this

```
output: {
    path: path.resolve(__dirname, 'dist'),
    filename: 'index.js',
    chunkFilename: '[name].js'
  },
```
then add require path up top like 

`const path = require('path');`

**Comment by jerme404** [comment]
There's a PR with my changes.

https://github.com/iam4x/pokemongo-webspoof/pull/436

**Comment by ascenser** [comment]
Is there a simplified fix for this yet for those of us who apparently aren't quite as tech-savvy? I've tried following the various tips in this thread but I'm getting no results, and can't even find some of the files referenced, like GoogleMap.spec.js

**Comment by ItaloFSS** [comment]
Thanks for the help. I got it to compile, added the new add to accessibility - but, I'm getting the following error in the app. Any help would be appreciated.

TypeError: Cannot read property 'maps' of undefined

TypeError: Cannot read property 'latLng' of undefined

**Comment by ItaloFSS** [comment]
jerme404, thanks for the help. I got it working.

ascenser if you need help, I can compile the app for you.

**Comment by CavemanGer** [comment]
I can't find the file where I have to put my generated key in. Programming is not my strength :-(

Is there anybody who could help me adding the key and rebuild the project?

Thanks in advance!

**Comment by htyx75** [comment]
I have the same issue, i got my own API key from google, but i am not a developer.
Is anyone able to help me via TeamViewer?
Help is really appreciated.

**Comment by twelvefiftyone** [comment]
@ItaloFSS  
could you please help compile the app?
Tried for an hour, still not working.

**Comment by jerme404** [comment]
I put a new build on Google Drive if you check the gitter link posted above. 

I suppose someone could make the api key configurable in the UI.

**Comment by htyx75** [comment]
@jerme404 , which link do you mean?

**Comment by ItaloFSS** [comment]
Here's the link https://www.dropbox.com/sh/gvrrvl8dkhzpior/AABN1dHcP0_o6lSPOZ111EL1a?dl=0

Get it now, I'm deleting the file tomorrow.

Good-luck.

**Comment by htyx75** [comment]
This folder is empty...

**Comment by ItaloFSS** [comment]
It was uploading. Is there now.

**Comment by htyx75** [comment]
Thanks, i got.
While running i get the following error:
<img width="325" alt="screen shot 2017-09-03 at 22 50 32" src="https://user-images.githubusercontent.com/31596986/30006524-5e310f7e-90fa-11e7-90c8-304759721ebe.png">

Any solution for this?

**Comment by ItaloFSS** [comment]
Under Security & Privacy > Accessibility add the app, Terminal, and Script Editor (the last two are found in the app utilities folder)

**Comment by htyx75** [comment]
Wonderful. Thx again

**Comment by BearinErin** [comment]
Thank you guys for the help! Works perfectly again.

**Comment by twelvefiftyone** [comment]
thanks a lot! @ItaloFSS 
will give another try from jerme404 advice as well.

**Comment by JonathanLu2393** [comment]
@yrchern That didn't work for me

**Comment by ry4meck** [comment]
So I downloaded git hub, all the source files and found where I needed to add my API in the source code text file. Can someone walk me through the process of compiling everything into a new app? Do I need another program. Sorry completely new at this.

**Comment by jnrepo** [comment]
@ry4meck 
```
$ git clone https://github.com/iam4x/pokemongo-webspoof.git
$ cd pokemongo-webspoof && npm install
$ npm run dev
```

**Comment by ry4meck** [comment]
@jnrepo where is that done from? from the github desktop app or is there something else I'm missing?

**Comment by jnrepo** [comment]
@ry4meck did you use terminal to clone the repository then make the code changes? Or did you download from releases?

You have to use the terminal to build the program after you make the code changes.

**Comment by ry4meck** [comment]
@jnrepo downloaded the releases to desktop. main folder is " pokemongo-webspoof-master". Then I found the file that had the API needed changed, changed it via text file and then saved.  That's basically where I am. So I guess I need to open terminal and figure how to import that and then build everything

**Comment by bgou** [comment]
there are 2 ways to change this:

### Preferred 

#### pokemongo-webspoof/src/config/api.js

Change the apiKey value to your own:
` apiKey: 'REPLACE_WITH_YOUR_KEY'`

### Other (Not preferred)

#### pokemongo-webspoof/src/views/map/index.js

Change
```
          <GoogleMap
            ...
            apiKey={ MapsApi.apiKey }>
```

to 

```
          <GoogleMap
            ...
            apiKey="YOUR_KEY">
```

**Comment by pospi12** [comment]
So I've been using this for a few months and today I got the error OVER_QUEY_LIMIT, after reading this I got my new API key, but I cannot find out how to add this on my iMac. I tried reading all the above info, but no idea where to find pokemongo-webspoof/src/config/api.js etc. Can somebody give me a walkthrough?

**Comment by stormdragon288** [comment]
Im in the same boat Pospi. I just got this issue now as well.

**Comment by jerme404** [comment]
@pospi12  From the root of the repository, it's in the path you just posted.

I don't think my fix was built into a release, so you'll need to download and build it yourself.  

<img width="1096" alt="screen shot 2018-04-23 at 6 29 52 pm" src="https://user-images.githubusercontent.com/29779465/39156548-9c918678-4724-11e8-8db2-7cda631cc608.png">

**Comment by stormdragon288** [comment]
@jerme404 I'm still having the issue after adding the config/api.js file.

**Comment by jerme404** [comment]
If all your doing is adding that file, it's not going to work.

**Comment by stormdragon288** [comment]
@jerme404 so you have to do is add the entire src folder? or all the files you just posted? Some directions would be nice. I already have my API key as well and replaced that in the file.

**Comment by stormdragon288** [comment]
@jerme404 Adding all files worked. Thanks again!

**Comment by jerme404** [comment]
Just git pull, and you'll have all of the changes you need.  Or just check my PR #436.  You have to actually build the project to implement my api key fix.
